### PR TITLE
Put columns into icons for BOMS

### LINKED ISSUE #1877: Make the links and notes sections action menu items with icons (separate from the ... dropdown)
### Description

Currently, the link and notes sections on a Bill of Material have their own column. Because having their own column doesn't provide any extra meaningful information, we want to make the link and notes section icons on the right (like how save and x look in the edit mode of a task).

### Acceptance Criteria

- [ ] The notes column on the BOM table should now be an action menu item on the right with a Notes MUI Icon
- [ ] The link column on the BOM table should now be an action menu item on the right side with a Link MUI Icon

### Proposed Solution

See how we add the action menu items that aren't in the ... on the task page

### PR DESCRIPTION
**Pull request by ryanhowe28**
## Changes

instead of a column, made notes and links into icons
## Screenshots

image
image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1877

### CODE REVIEW

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx [hunk 1]
```diff
@@ -243,6 +243,7 @@ const TaskListDataGrid: React.FC<TaskListDataGridProps> = ({
             icon={<DescriptionIcon fontSize="small" />}
             label="notes"
             onClick={() => {
+              console.log('hitting this button');
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx:246]: fart

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
we should make the title of the column actions it looks not great with no title

also on medium to small screens they dissapear they should be combined into one icon

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx (hunk 1 above)
Comment (ryanhowe28) [reply src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskList/TaskListDataGrid.tsx:246]: lmaoooo

**Review by Peyton-McKee — APPROVED** [review]
noice

**Pull request merged**